### PR TITLE
chore(Data/Vector): reduce autoImplicit

### PR DESCRIPTION
**Pull request by grunweg**
After this PR, `autoImplicit` in this directory is limited to three declarations.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Vector/MapLemmas.lean [hunk 1]
```diff
@@ -56,10 +56,12 @@ theorem map_map (f₁ : β → γ) (f₂ : α → β) :
...
 
 section Binary
-variable (xs : Vector α n) (ys : Vector β n)
+
+variable {α β : Type} {n : ℕ} (xs : Vector α n) (ys : Vector β n)
+variable {σ₁ σ₂ : Type} {s₁ : σ₁} {s₂ : σ₂}
 
 @[simp]
-theorem mapAccumr₂_mapAccumr_left (f₁ : γ → β → σ₁ → σ₁ × ζ) (f₂ : α → σ₂ → σ₂ × γ) :
+theorem mapAccumr₂_mapAccumr_left {γ ζ : Type} (f₁ : γ → β → σ₁ → σ₁ × ζ) (f₂ : α → σ₂ → σ₂ × γ) :
```
Comment (fpvandoorn) [inline Mathlib/Data/Vector/MapLemmas.lean:64]: Is the missing `*` intentional (also below and above)?
Please make these variables.

Comment (grunweg) [reply Mathlib/Data/Vector/MapLemmas.lean:64]: If you look very closely, you'll see that all lemmas in this section alternatingly have `Type` and `Type*`: the way they are currently written, this is the best one can do. I could extract the Type* case as variables (highlighting the non-polymorphic case); would you prefer this?

**Comment by grunweg** [comment]
awaiting-review

File: Mathlib/Data/Vector/MapLemmas.lean (hunk 1 above)
Comment (riccardobrasca) [reply Mathlib/Data/Vector/MapLemmas.lean:64]: Does something really break with `Type*` everywhere?

Comment (grunweg) [reply Mathlib/Data/Vector/MapLemmas.lean:64]: I've tried this in [this branch](https://github.com/leanprover-community/mathlib4/compare/MR-vector-typestart): six new errors about "unsolved goals" appear. Be my guest to PR and fix them - I'd like to declare this out of scope of this PR.

Comment (grunweg) [reply Mathlib/Data/Vector/MapLemmas.lean:64]: I have pushed a commit, however, which makes Type* the common case - to emphasize the `Type` is unusual.

Comment (riccardobrasca) [reply Mathlib/Data/Vector/MapLemmas.lean:64]: `Type*` means `Type u` for a new universe variable `u`, where `Type` means `Type 1`: in particular `Type*` is strictly more general than `Type`.

File: Mathlib/Data/Vector/Basic.lean [hunk 2]
```diff
@@ -814,16 +813,18 @@ theorem get_map₂ (v₁ : Vector α n) (v₂ : Vector β n) (f : α → β →
     · simp only [get_zero, head_cons]
     · simp only [get_cons_succ, ih]
 
+set_option autoImplicit true in
```
Comment (riccardobrasca) [inline Mathlib/Data/Vector/Basic.lean:816]:
```lean
@[simp]
theorem mapAccumr_cons {α : Type*} (xs : Vector α n) :
    ∀ {α₁ : Type} {β : Type*} {f : α → α₁ → α₁ × β} {x : α} {s : α₁},
    mapAccumr f (x ::ᵥ xs) s
    = let r := mapAccumr f xs s
      let q := f x r.1
      (q.1, q.2 ::ᵥ r.2) :=
  rfl
```
this seems to work. I guess the next one is similar.

**Comment by grunweg** [comment]
#16154 solves this part in a much nicer way: let me close this PR in favour of it.

**Pull request closed without merging**